src/custom-element.js

```diff
@@ -591,13 +592,145 @@ function createBaseCustomElementClass(win, elementConnectedCallback) {
+     *
+     * Can only be called on a upgraded element. May only be called from
+     * scheduler.js.
+     *
+     * @return {!Promise}
+     * @final
+     * @restricted
+     */
+    mountInternal() {
+      if (this.mountPromise_) {
+        return this.mountPromise_;
+      }
+      this.mountAbortController_ = new AbortController();
+      const {signal} = this.mountAbortController_;
+      return (this.mountPromise_ = this.buildInternal()
+        .then(() => {
+          devAssert(this.V1());
+          if (signal.aborted) {
+            // Mounting has been canceled.
+            return;
+          }
+          this.setReadyStateInternal(
+            this.implClass_.load(this)
+              ? ReadyState.LOADING
+              : ReadyState.MOUNTING
+          );
+          return this.impl_.mountCallback(signal);
+        })
+        .then(() => {
+          this.mountAbortController_ = null;
+          if (signal.aborted) {
+            throw cancellation();
+          }
+          this.signals_.signal(CommonSignals.MOUNTED);
+          if (
+            this.implClass_.load(this) &&
+            this.readyState_ !== ReadyState.COMPLETE
+          ) {
+            this.setReadyStateInternal(ReadyState.LOADING);
+          } else {
+            this.setReadyStateInternal(ReadyState.COMPLETE);
+          }
+        })
+        .catch((reason) => {
+          this.mountAbortController_ = null;
+          if (isCancellation(reason)) {
+            this.mountPromise_ = null;
+          } else {
+            this.signals_.rejectSignal(
+              CommonSignals.MOUNTED,
+              /** @type {!Error} */ (reason)
+            );
+            this.setReadyStateInternal(ReadyState.ERROR, reason);
+          }
+          throw reason;
+        }));
+    }
+
+    /**
+     * Requests the element to be mounted as soon as possible.
+     * @return {!Promise}
+     * @final
+     */
+    mount() {
+      if (this.mountPromise_) {
+        return this.mountPromise_;
+      }
+      return this.build().then(() => {
+        devAssert(this.V1());
+        const scheduler = getSchedulerForDoc(this.getAmpDoc());
+        scheduler.scheduleAsap(this);
+        return this.whenMounted();
+      });
+    }
+
+    /**
+     * Unmounts the element and makes it ready for the next mounting
+     * operation.
+     * @final
+     */
+    unmount() {
+      // Hasn't been mounted yet.
+      if (!this.mountPromise_) {
+        return;
+      }
+
+      devAssert(this.V1());
+
+      // Cancel the currently mounting operation.
+      if (this.mountAbortController_) {
+        this.mountAbortController_.abort();
+        this.mountAbortController_ = null;
+      }
+
+      if (this.isConnected_) {
+        // Ensure that the element is paused.
+        this.pause();
+      }
+
+      // Try to unmount. Not every element can be unmounted or has anything
+      // to unmount.
+      const unmounted = this.impl_.unmountCallback();
+      if (!unmounted) {
```

Comment:
Should we check this before calling `mountAbortController_.abort()`?

Reply:
Interesting. But then it'd mean that `unmountCallback` has to be called before the abort signal is sent? I think this is a weird dependency. Basically we have to say: if mount is abortable then it also means that it must be reentrant.